src/plot_api/subroutines.js

```diff
@@ -0,0 +1,339 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../plotly');
+var Registry = require('../registry');
+var Plots = require('../plots/plots');
+var Lib = require('../lib');
+
+var Color = require('../components/color');
+var Drawing = require('../components/drawing');
+var Titles = require('../components/titles');
+var ModeBar = require('../components/modebar');
+
+
+exports.layoutStyles = function(gd) {
```

Comment:
Attaching plot/update subroutines to a separate modules allows us to easily test whether the correct subroutine is called on update using jasmine spies. See https://github.com/plotly/plotly.js/pull/875/commits/26c407597c22d4ba749b06d70a40e0ce3db3f6db for example.